internal/manifests/manifestutils/resources.go

```diff
@@ -49,6 +49,9 @@ func Resources(tempo v1alpha1.TempoStack, component string) corev1.ResourceRequi
 	if ok {
 		totalCpuInt := totalCpu.MilliValue()
 		cpu := float32(totalCpuInt) * componentResources.cpu
+		if r := tempo.Spec.Template.Compactor.Replicas; r != nil && *r > 1 {
```

Comment:
`Resources()` is called from every tempo component (compactor, distributor, ingester, etc). I'd suggest to add an additional function parameter `replicas *int32` to `Resources()` and update all callers.

Reply:
> Resources() is called from every tempo component (compactor, distributor, ingester, etc).

I see, but would there be any other possibilities for those components to have a replicas value other than `tempo.Spec.Template.Compactor.Replicas`?

because what I imagined is something like this

```
--- Resources:       manifestutils.Resources(tempo, manifestutils.CompactorComponentName),
+++ Resources:       manifestutils.Resources(tempo, manifestutils.CompactorComponentName, tempo.Spec.Template.Compactor.Replicas)
```

(file:  [compactor.go](https://github.com/grafana/tempo-operator/blob/147be1594663788b337d3ea9327f0543b18160e4/internal/manifests/compactor/compactor.go#L111))

Reply:
Yes, each component can have individual replica settings - e.g. you can have 2 replicas of the compactor and 10 of the distributor.

https://github.com/grafana/tempo-operator/blob/505a7ebb0aa70dc84002b171c79b62fae33c7a54/internal/manifests/distributor/distributor.go#L197
https://github.com/grafana/tempo-operator/blob/505a7ebb0aa70dc84002b171c79b62fae33c7a54/internal/manifests/ingester/ingester.go#L66
etc.

Reply:
i see, thanks for the pointers

i will do the update

Reply:
updated, ready to review